src/linkgraph/mcf.cpp

```diff
@@ -518,8 +523,10 @@ MCF1stPass::MCF1stPass(LinkGraphJob &job) : MultiCommodityFlow(job)
 							path->GetFreeCapacity() > INT_MIN) {
 						this->PushFlow(edge, path, accuracy, UINT_MAX);
 					}
+					if (edge.UnsatisfiedDemand() > 0) source_demand_left = true;
 				}
 			}
+			if (!source_demand_left) finished_sources[source] = true;
```

Comment:
could avoid the branch with
```suggestion
			finished_sources[source] = !source_demand_left;
```